File: Codigos estudiantes por lenguaje/Java/Cristian Maldonado/Jump_Search.java
public class JumpSearch {

	public static void main(String[] args) {
		int arr[] = { 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610 };
		int x = 5;
		int resultado = jumpSearch(arr, x);
		if (resultado == -1) {
			System.out.println("no encontrado");
		} else {
			System.out.println("Encontrado en el indice " + resultado);
		}
	}

	static int jumpSearch(int arr[], int x) {
		int n = arr.length;
		int step = (int) Math.floor(Math.sqrt(n));
		int prev = 0;
		while (arr[Math.min(step, n) - 1] < x) {
			prev = step;
			step += (int) Math.floor(Math.sqrt(n));
			if (prev >= n) {
				return -1;
			}
		}
		while (arr[prev] < x) {
			prev++;
			if (prev == Math.min(step, n)) {
				return -1;
			}
		}
		if (arr[prev] == x) {
			return prev;
		}
		return -1;
	}
}

File: Codigos estudiantes por lenguaje/C++/Diana Molares/introduccion/jumpSearch.cpp
#include <bits/stdc++.h>
#include <cstdlib>
using namespace std;
static int jumpSearch(int arr[],int x,int n){
	int step=int(std::floor(std::sqrt(n)));
	int prev =0;
	while(arr[std::min(step,n)-1]<x){
		prev=step;
		step+=int(std::floor(std::sqrt(n)));
		if(prev>=n){
			return -1;
		}
	}
	while(arr[prev]<x){
		prev++;
		if(prev==std::min(step,n)){
			return -1;
		}
	}
	if(arr[prev]==x){
		return prev;
	}
	return -1;
			
}
int main(int argc, char const *argv[]) {
    int arr[] = {1, 2, 3, 4, 5};
    int n = sizeof (arr) / sizeof (arr[0]);
    int x=5;
    int result = jumpSearch(arr, x,n);
    if (result < 0) {
        cout << "El elemento no esta presente en el Array" << endl;
    } else {
        cout << "Elemento encontrado en el indice " << result << endl;
    }

    return 0;
}
